src/test/java/edu/unc/lib/boxc/migration/cdm/services/FindingAidReportServiceTest.java

```diff
@@ -0,0 +1,113 @@
+    private FindingAidReportService service;
+    private SipServiceHelper testHelper;
+
+    private AutoCloseable closeable;
+
+
+    @BeforeEach
+    public void setup() throws Exception {
+        closeable = openMocks(this);
+        project = MigrationProjectFactory.createMigrationProject(
+                tmpFolder, PROJECT_NAME, null, "user", CdmEnvironmentHelper.DEFAULT_ENV_ID,
+                BxcEnvironmentHelper.DEFAULT_ENV_ID, MigrationProject.PROJECT_SOURCE_CDM);
+        Files.createDirectories(project.getExportPath());
+
+        basePath = tmpFolder.resolve("testFolder");
+        Files.createDirectory(basePath);
+        testHelper = new SipServiceHelper(project, basePath);
+
+        service = new FindingAidReportService();
+        service.setProject(project);
+        service.setIndexService(testHelper.getIndexService());
+    }
+
+    @Test
+    public void descriFieldReportTest() throws Exception {
+        testHelper.indexExportData("03883");
+        FindingAidReportOptions options = new FindingAidReportOptions();
+        options.setField("descri");
+
+        service.fieldCountUniqueValuesReport(options);
+
+        assertTrue(Files.exists(service.getFieldValuesReportPath(options)));
+        try (CSVParser csvParser = parser(service.getFieldValuesReportPath(options))) {
+            List<CSVRecord> rows = csvParser.getRecords();
+            assertIterableEquals(Arrays.asList("descri_value", "count"), rows.get(0));
+            assertIterableEquals(Arrays.asList("03883", "2"), rows.get(1));
+            assertEquals(2, rows.size());
+        }
+    }
+
+    @Test
+    public void contriFieldReportTest() throws Exception {
+        testHelper.indexExportData("mini_gilmer");
+        FindingAidReportOptions options = new FindingAidReportOptions();
+        options.setField("contri");
+
+        service.fieldCountUniqueValuesReport(options);
+
+        assertTrue(Files.exists(service.getFieldValuesReportPath(options)));
+        try (CSVParser csvParser = parser(service.getFieldValuesReportPath(options))) {
+            List<CSVRecord> rows = csvParser.getRecords();
+            assertIterableEquals(Arrays.asList("contri_value", "count"), rows.get(0));
+            assertIterableEquals(Arrays.asList("Fitz, Newton.", "2"), rows.get(1));
+            assertIterableEquals(Arrays.asList("(blank)", "1"), rows.get(2));
+            assertEquals(3, rows.size());
+        }
+    }
+
+    @Test
+    public void hookIdReportTest() throws Exception {
+        testHelper.indexExportData("03883");
+
+        service.hookIdReport();
+
+        assertTrue(Files.exists(service.getHookIdReportPath()));
+        try (CSVParser csvParser = parser(service.getHookIdReportPath())) {
+            List<CSVRecord> rows = csvParser.getRecords();
+            assertIterableEquals(Arrays.asList("hookid", "count"), rows.get(0));
+            assertIterableEquals(Arrays.asList("03883_folder_5", "2"), rows.get(1));
```

Comment:
It looks like the 03883 collection only has 2 records and they both should have different hook ids, since they have contri values `folder_5` and `folder_9`, so I think the assertion here is not correct at the moment. I think it relates to my suggest back up in the FindingAidReportService and grouping. I would suggest extending 03883 so that there are actually two entries with the same hookid, and at least one entry with a different hookid so that we can verify it differentiates